### PR TITLE
[RLlib][Docs] Enable doctests for RLLib

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

#35286 temporarily disabled doctests for RLLib because there are many broken and (previously) untested docstring examples. This PR re-enables them.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/apex_ddpg/apex_ddpg.py [hunk 1]
```diff
@@ -14,32 +14,38 @@ class ApexDDPGConfig(DDPGConfig):
...
+
+        .. code-block:: python
+
+            from ray.rllib.algorithms.apex_ddpg.apex_ddpg import ApexDDPGConfig
+            from ray import tune
+            import ray.air as air
+            config = ApexDDPGConfig()
+            # Print out some default values.
+            print(config.lr) # doctest: +SKIP
+            0.0004
```
Comment (avnishn) [inline rllib/algorithms/apex_ddpg/apex_ddpg.py:37]: I think this line was mean to by the output of the above print statement, but probably doesn't belong in the codeblock, right?

File: rllib/algorithms/simple_q/simple_q.py [hunk 2]
```diff
@@ -55,8 +55,8 @@ class SimpleQConfig(AlgorithmConfig):
         >>>         "capacity":  40000,
         >>>     }
         >>> )
-        >>> config.training(replay_buffer_config=replay_config)\
-        ...       .resources(num_gpus=1)\
+        >>> config.training(replay_buffer_config=replay_config)
```
Comment (avnishn) [inline rllib/algorithms/simple_q/simple_q.py:58]: when should we use >>> vs code blocks?

**Review by avnishn — COMMENTED** [review]
I have 1 question and 1 comment

File: rllib/algorithms/apex_ddpg/apex_ddpg.py (hunk 1 above)
Comment (bveeramani) [reply rllib/algorithms/apex_ddpg/apex_ddpg.py:37]: Ah, good catch. Fixed

File: rllib/algorithms/simple_q/simple_q.py (hunk 2 above)
Comment (bveeramani) [reply rllib/algorithms/simple_q/simple_q.py:58]: See https://docs.ray.io/en/master/ray-contribute/writing-code-snippets.html#which-type-of-example-should-you-write.

Comment (bveeramani) [reply rllib/algorithms/simple_q/simple_q.py:58]: tl;dr: Never use `code-block:: python`. If you really want to emphasize object representations, use `>>>`. Most of the time you probably want `testcode`, though.

File: rllib/algorithms/apex_ddpg/apex_ddpg.py [hunk 3]
```diff
@@ -14,32 +14,36 @@ class ApexDDPGConfig(DDPGConfig):
...
 
     Example:
-        >>> from ray.rllib.algorithms.apex_ddpg.apex_ddpg import ApexDDPGConfig
-        >>> config = ApexDDPGConfig().training(lr=0.01).resources(num_gpus=1)
-        >>> print(config.to_dict()) # doctest: +SKIP
-        >>> # Build a Trainer object from the config and run one training iteration.
-        >>> algo = config.build(env="Pendulum-v1")
-        >>> algo.train()  # doctest: +SKIP
+
+        .. code-block:: python
```
Comment (avnishn) [inline rllib/algorithms/apex_ddpg/apex_ddpg.py:18]: wait weren't you saying to never use code-block:: python?

**Review by avnishn — APPROVED** [review]
conditionally approving, but I have one question about your usage of code-block:: python in this pr.

**Comment by bveeramani** [comment]
Failing doctest is unrelated.

```
086     import ray
--
  | 087
  | 088     # Add this at the top of your Ray application.
  | 089     runtime_env = {"env_vars": {"NCCL_SOCKET_IFNAME": "ens5"}}
  | 090     ray.init(runtime_env=runtime_env, ignore_reinit_error=True)
```

**Pull request merged**